test/test_helper.rb

```diff
@@ -110,6 +113,58 @@ def fake_machine(options = {})
   machine
 end
 
+class FakeHTTPProxyServer < WEBrick::HTTPProxyServer
+  def initialize(url, config)
+    @url = url
+    super(config)
+  end
+
+  def service(req, res)
+    req.attributes['url'] = @url
+    super(req, res)
+  end
+end
+
+module WEBrick
```

Comment:
Is this not a matter of how you configure your logging? It seems it defaults to stdout. Just provide a logger. Either another StringIO or even the same. Monkey patching the log method seems bad.